src/fn/upload-docs.js

```diff
@@ -93,20 +97,92 @@ const execute = async () => {
     }
   };
 
-  return processNextBatch(filesToUpload, INITIAL_BATCH_SIZE);
-};
+  return processNextBatch(filenamesToUpload, INITIAL_BATCH_SIZE);
+}
+
+function warnAndPrompt(warningMessage) {
+  warn(warningMessage);
+  if (!userPrompt.keyInYN()) {
+    throw new Error('User aborted execution.');
+  }
+}
 
-const getErrorsWhereDocIdDiffersFromFilename = filePaths =>
-  filePaths
+function analyseFiles(filePaths) {
+  return filePaths
     .map(filePath => {
       const json = fs.readJson(filePath);
       const idFromFilename = path.basename(filePath, FILE_EXTENSION);
 
       if (json._id !== idFromFilename) {
-        return `File '${filePath}' sets _id:'${json._id}' but the file's expected _id is '${idFromFilename}'.`;
+        return { error: `File '${filePath}' sets _id:'${json._id}' but the file's expected _id is '${idFromFilename}'.` };
+      }
+
+      if (json._deleted) {
+        return { delete: json._id };
       }
     })
-    .filter(err => err);
+    .filter(Boolean);
+}
+
+async function handleUsersAtDeletedFacilities(deletedDocIds) {
+  const affectedUsers = await getAffectedUsers(deletedDocIds);
+  const usernames = affectedUsers.map(userDoc => userDoc.username).join(', ');
+  if (affectedUsers.length === 0) {
+    return;
+  }
+
+  warnAndPrompt(`This operation will update permissions for ${affectedUsers.length} user accounts: ${usernames}. Are you sure you want to continue?`);
+  await updateAffectedUsers(affectedUsers);
+}
+
+
+async function getAffectedUsers(deletedDocIds) {
```

Comment:
suggestion: if `deletedDocIds` are sure to be `facilityIds` then, renaming it to `facilityIds` makes more sense

Reply:
sadly, at this point we can't guarantee they are facilities. app devs write many types of scripts which create documents and dump them into the `json_docs` folder to be uploaded by the `upload-docs` action.  

i don't think the paramater should be specific to facilities at this point... this function is kinda _checking if the deleted doc ids end up being user's facility_ids (affectedUsers)_